src/main/java/org/opensearch/neuralsearch/processor/chunker/ChunkerParameterValidator.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.chunker;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+
+import java.util.Map;
+import java.util.Locale;
+
+/**
+ * Validate the parameter for text chunking processor and algorithms
+ */
+public class ChunkerParameterValidator {
+
+    /**
+     * Validate string type parameter
+     */
+    public static void validateStringParameter(final Map<String, Object> parameters, final String fieldName, final boolean allowEmpty) {
+        if (!parameters.containsKey(fieldName)) {
+            // all chunking algorithm parameters are optional
+            return;
+        }
+        Object fieldValue = parameters.get(fieldName);
+        if (!(fieldValue instanceof String)) {
+            throw new IllegalArgumentException(
+                String.format(Locale.ROOT, "Parameter [%s] cannot be cast to [%s]", fieldName, String.class.getName())
+            );
+        }
+        if (!allowEmpty && StringUtils.isEmpty(fieldValue.toString())) {
+            throw new IllegalArgumentException(String.format(Locale.ROOT, "Parameter [%s] should not be empty.", fieldName));
+        }
+    }
+
+    /**
+     * Validate integer type parameter with positive value
+     */
+    public static void validatePositiveIntegerParameter(
+        final Map<String, Object> parameters,
+        final String fieldName,
+        final int defaultValue
+    ) {
+        if (!parameters.containsKey(fieldName)) {
+            // all chunking algorithm parameters are optional
+            return;
+        }
+        String fieldValueString = parameters.get(fieldName).toString();
+        if (!(NumberUtils.isParsable(fieldValueString))) {
```

Comment:
We can remove this isParsable method and add try...catch to `NumberUtils.createInteger(fieldValueString)`, because we expect integer here and isParsable is to generic.

Reply:
Already updated.